app/controllers/admin/discourse_calendar/admin_holidays_controller.rb

```diff
@@ -1,19 +1,39 @@
 # frozen_string_literal: true
 
-require "holidays"
-
 module Admin::DiscourseCalendar
   class AdminHolidaysController < AdminDiscourseCalendarController
     def index
       region_code = params[:region_code]
 
       begin
-        holidays = Holidays.year_holidays([region_code], Time.current.beginning_of_year)
+        holidays = DiscourseCalendar::Holiday.find_holidays_for(region_code: region_code)
       rescue Holidays::InvalidRegion
         return render_json_error(I18n.t("system_messages.discourse_calendar_holiday_region_invalid"), 422)
       end
 
       render json: { region_code: region_code, holidays: holidays }
     end
+
+    def disable
+      DiscourseCalendar::DisabledHoliday.create(disabled_holiday_params)
```

Comment:
We should validate the parameters before saving them in case there's a bug in the client.

Reply:
I added some Rails `presence` validations for `holiday_name` and `region_code`, I also updated the migration to make sure they're not null.

Reply:
I also added a test case for when the parameters are missing.

Reply:
That's great! Should we also check for invalid region codes? I assume the holiday name could be anything, or should we validate that against the gem holidays too?

Reply:
I don't think it'll be too difficult to validate the region code against the Holidays gem, I can work on that. :+1:  Maybe we could even do the holiday names too, I'll check that out too!

Reply:
Looks great now. Thanks!